### PR TITLE
fix panic on broken symlink in git repository

### PR DESCRIPTION
**Pull request by msehnout**
The issue including reproducer is described here:
https://github.com/ogham/exa/issues/526
This commit includes proposed change as well as integration test.

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
should also fix https://github.com/ogham/exa/issues/558

now I have no idea how to test this but could this be merged asap ?? @ogham it's critical

**Comment by clarfon** [comment]
Any plans to merge this soon?

**Comment by flokli** [comment]
Ping @ogham - would be nice to see this fixed.

**Comment by MugunthKumar** [comment]
I had this problem too. All good after deleting the offending broken symlink. Great to have this merged!

**Comment by BarbzYHOOL** [comment]
:'(

**Comment by ogham** [comment]
Thank you for this fix!

Apologies for taking so long to merge it.

File: src/fs/feature/git.rs [hunk 1]
```diff
@@ -265,11 +265,11 @@ impl Git {
 fn reorient(path: &Path) -> PathBuf {
     use std::env::current_dir;
     // I’m not 100% on this func tbh
-    match current_dir() {
+    let path = match current_dir() {
         Err(_)  => Path::new(".").join(&path),
         Ok(dir) => dir.join(&path),
-    }.canonicalize().unwrap()   // errors can be ignored here because they only occur if
```
Comment (snipem) [inline src/fs/feature/git.rs:271]: So far with assumptions :) Good fix

**Comment by Holzhaus** [comment]
I stumble over this quite often. Could you tag a stable release including this fix? That would be great!

**Pull request merged**